pkg/controller/cyclenoderequest/transitioner/transitions.go

```diff
@@ -170,6 +170,60 @@ func (t *CycleNodeRequestTransitioner) transitionPending() (reconcile.Result, er
 		}
 	}
 
+	if t.cycleNodeRequest.Spec.HealthChecks != nil {
```

Comment:
I think this logic should be broken out into a discrete function

Reply:
Good point. The phases are getting a bit big